### ISSUE TITLE
demo example error

### ISSUE THREAD

**Issue by nottwy** [duct317/scDHA#1]
Dear developers,

I have correctly install your software. Then I try to run your following demo codes.

==================================
1. Load the package: library(scDHA)
2. Load SingleCellExperiment package to read dataset: library(SingleCellExperiment)
3. Load Goolam dataset: data('Goolam'); data <- t(assay(Goolam)); label <- Goolam$cell_type1
4. Log transform the data: data <- log2(data + 1)
5. Generating clustering result: result <- scDHA(data, seed = 1)
==================================
step 1 - 4 report no error.
For step 5, I get such an error:
> result <- scDHA(data, seed = 1)
Error in { : task 1 failed - "argument is of length zero"

Could you help me?
Thanks a lot!

**Comment by nottwy** [comment]
Error in {: task 1 failed - "argument is of length zero"
Traceback:

1. scDHA(data, seed = 1)
2. scDHA.small(data = data, k = k, K = K, n = n, ncores = ncores, 
 .     gen_fil = gen_fil, do.clus = do.clus, sample.prob = sample.prob, 
 .     seed = seed)
3. gene.filtering(data.list = data.list, original_dim = original_dim, 
 .     batch_size = batch_size, ncores.ind = ncores.ind, wdecay = wdecay, 
 .     seed = seed)
4. foreach(i = seq(3)) %dopar% {
 .     if (is.null(seed)) {
 .         config <- list()
 .         config$intra_op_parallelism_threads <- ncores.ind
 .         config$inter_op_parallelism_threads <- ncores.ind
 .         session_conf <- do.call(tf$ConfigProto, config)
 .         sess <- tf$Session(graph = tf$get_default_graph(), config = session_conf)
 .         k_set_session(session = sess)
 .     }
 .     else {
 .         set.seed((seed + i))
 .         use_session_with_seed((seed + i))
 .     }
 .     data.tmp <- data.list[[i]]
 .     batch_size <- round(nrow(data.tmp)/50)
 .     x <- layer_input(shape = c(original_dim))
 .     h <- layer_dense(x, 50, kernel_constraint = constraint_nonneg())
 .     x_decoded_mean <- layer_dense(h, original_dim)
 .     vae <- keras_model(x, x_decoded_mean)
 .     vae %>% compile(optimizer = tf$contrib$opt$AdamWOptimizer(wdecay, 
 .         0.001), loss = "mse")
 .     his <- vae %>% fit(data.tmp, data.tmp, shuffle = TRUE, epochs = 10, 
 .         batch_size = batch_size, verbose = 0)
 .     W <- get_weights(get_layer(vae, index = 2))[[1]]
 .     Wsd <- rowSds(W)
 .     Wsd[is.na(Wsd)] <- 0
 .     Wsd <- (Wsd - min(Wsd))/(max(Wsd) - min(Wsd))
 .     Wsd
 . }
5. e$fun(obj, substitute(ex), parent.frame(), e$data)

**Comment by duct317** [comment]
Hi,

Before the package need to call Tensorflow from Python to build and train the neural network, it could lead to some unexpected errors. We just update our package to change to Torch, which can be called directly from R. The developing version can be found here: https://github.com/duct317/scDHA/tree/dev-torch
It can be installed using (CRAN version will be available soon): 
`install.packages("RcppAnnoy")`
`devtools::install_github('duct317/scDHA@dev-torch')`
It should be much easier to install. Can you try this new version? Note that the instruction to load demo data change a little bit as we remove the need for `SingleCellExperiment` package. Please let me know if you still have problem.

**Comment by nottwy** [comment]
Thank you for your light-speed response!
It's really great that you have updated your software.
Before I try your new version, I want to ask you one really important thing.
Which version's performance is better?This latest version or the old one? Or their performance is similar?

**Comment by duct317** [comment]
For accuracy, when we benchmark using the datasets in the original paper, the two version have similar performance. The new version is a little bit better, I think ARI 0.01 higher in average.
For speed, due to the limitations of R, e.g. slower BLAS library and in general slower than python, the running time is around 1.5-2 times slower. But if you can switch to  OpenBLAS or something similar, then they are similar again.
https://csantill.github.io/RPerformanceWBLAS/
